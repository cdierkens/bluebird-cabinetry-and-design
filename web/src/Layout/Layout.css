/* purgecss start ignore */
@tailwind base;
@tailwind components;

/* purgecss end ignore */

@tailwind utilities;

html {
  @apply font-body leading-normal;

  font-size: 16px;
}

body {
  @apply bg-white text-black;

  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code,
pre {
  @apply font-mono;
}

h1,
h2,
h3,
h4,
h5,
h6 {
  @apply font-sans font-bold text-blue-dark mt-4 mb-6;
}

p {
  @apply text-base mt-4 mb-6;
}

h1 {
  @apply text-5xl;
}

h2 {
  @apply text-4xl;
}

h3 {
  @apply text-3xl;
}

h4 {
  @apply text-2xl;
}

h5 {
  @apply text-xl;
}

h6 {
  @apply text-lg;
}

a:hover,
a:active,
a:focus {
  @apply text-gold;
}
