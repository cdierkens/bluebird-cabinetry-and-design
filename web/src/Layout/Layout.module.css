.Layout {
  /* Plopped styles for Layout */
}
