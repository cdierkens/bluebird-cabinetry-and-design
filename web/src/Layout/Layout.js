import PropTypes from "prop-types";
import React from "react";
import Footer from "./Footer";
import GraphQLErrors from "./GraphQLErrors";
import Head from "./Head";
import Header from "./Header";
import "./Layout.css";
import styles from "./Layout.module.css";

const Layout = ({ errors, children, ...props }) =>
  errors ? (
    <GraphQLErrors errors={errors} />
  ) : (
    <React.Fragment>
      <Head {...props} />

      <div className={styles.Layout}>
        <Header />

        <main>{children}</main>

        <Footer></Footer>
      </div>
    </React.Fragment>
  );

Layout.propTypes = {
  errors: PropTypes.arrayOf(
    PropTypes.shape({
      message: PropTypes.string,
    })
  ),
  children: PropTypes.node,
};

export default Layout;
