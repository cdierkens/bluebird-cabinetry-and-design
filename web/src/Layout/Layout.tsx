import React from "react";
import PageTitle from "../components/PageTitle";
import { todo } from "../migration.types";
import Footer from "./Footer";
import GraphQLErrors from "./GraphQLErrors";
import Head from "./Head";
import Header from "./Header";
import "./Layout.css";

interface Props {
  errors?: todo;
  hidePageTitle?: boolean;
  title: string;
}

const Layout: React.FC<Props> = ({
  errors,
  title,
  hidePageTitle,
  children,
  ...props
}) =>
  errors ? (
    <GraphQLErrors errors={errors} />
  ) : (
    <React.Fragment>
      <Head title={title} {...props} />

      <a href="#content" className="sr-only">
        Skip to content
      </a>

      <div>
        <Header />

        <main role="main" id="content">
          {!hidePageTitle && <PageTitle>{title}</PageTitle>}

          {children}
        </main>

        <Footer></Footer>
      </div>
    </React.Fragment>
  );

export default Layout;
