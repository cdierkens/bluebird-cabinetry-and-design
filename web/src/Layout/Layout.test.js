import { render } from "@testing-library/react";
import React from "react";
import Layout from "./Layout";

it("matches snapshot", () => {
  const { asFragment } = render(<Layout />);

  expect(asFragment()).toMatchSnapshot();
});
