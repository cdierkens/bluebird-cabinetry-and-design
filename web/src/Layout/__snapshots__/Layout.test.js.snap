// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="sr-only"
    href="#content"
  >
    Skip to content
  </a>
  <div>
    <header
      class="Header"
    >
      <nav
        class="DesktopNav grid-cols-6 lg:grid-cols-8"
      >
        <div
          class="HeaderLogo col-span-1 lg:col-span-2"
        >
          <a
            href="/"
          >
            <img
              alt="Bluebird Logo"
              class="Logo "
              src="test-file-stub"
            />
          </a>
        </div>
        <div
          class="HeaderNav col-span-5 lg:col-span-6"
        >
          <a
            class="Link"
            href="/"
          >
            Home
          </a>
          <a
            class="Link"
            href="/about"
          >
            About
          </a>
          <a
            class="Link"
            href="/services"
          >
            Services
          </a>
          <a
            class="Link"
            href="/portfolio"
          >
            Portfolio
          </a>
          <a
            class="Link"
            href="/contact"
          >
            Contact
          </a>
        </div>
      </nav>
    </header>
    <main
      id="content"
      role="main"
    >
      <h1
        class="PageTitle"
      >
        Test Title
      </h1>
    </main>
    <div
      class="bg-gray-darker text-center text-gray-light p-5 text-sm leading-loose"
    >
      <div
        class="my-6"
      >
        <test-file-stub
          classname="p-1 fill-current stroke-current inline-block h-24 text-white stroke-none"
        />
      </div>
      <div
        class="my-6"
      >
        <div>
          <a
            class="underline"
            href="tel:1-513-668-4529"
          >
            (513) 668‐4529
          </a>
        </div>
        <div>
          <a
            class="underline text-sm sm:text-base"
            href="mailto:dana@bluebirdcabinetryanddesign.com"
          >
            dana@bluebirdcabinetryanddesign.com
          </a>
        </div>
        <div>
          <span
            class="text-sm sm:text-base"
          >
            Serving the 
            <a
              class="underline"
              href="https://en.wikipedia.org/wiki/Cincinnati_metropolitan_area"
            >
              Greater Cincinnati Ohio Tri-State Area
            </a>
          </span>
        </div>
      </div>
      <div
        class="my-6 flex items-center justify-center flex-wrap"
      >
        <a
          class="mx-2"
          href="https://www.facebook.com/kitchenandbathdesigner/"
        >
          <test-file-stub
            classname="p-1 fill-current stroke-current inline-block h-12 text-white"
          />
          <span
            class="sr-only"
          >
            Facebook
          </span>
        </a>
        <a
          class="mx-2 underline"
          href="/trade-services"
        >
          To The Trade Services
        </a>
        <button
          class="ButtonWhite focus:outline-none focus:ring mb-5 mx-5 my-6"
          type="button"
        >
          Contact Us
        </button>
      </div>
      <div
        class="my-6 flex items-center justify-center flex-wrap"
      >
        <a
          class="mx-2"
          href="https://nkba.org/dana-snyder"
        >
          <test-file-stub
            classname="p-1 fill-current stroke-current inline-block h-10 text-white"
          />
          <span
            class="sr-only"
          >
            National Kitchen and Bath Association
          </span>
        </a>
        <test-file-stub
          classname="p-1 fill-current stroke-current inline-block mx-2 h-20 text-white"
        />
        <a
          class="mx-2 px-3"
          href="https://www.houzz.com/professionals/kitchen-and-bath-designers/dana-snyder-ckd-pfvwus-pf~1099041723"
        >
          <test-file-stub
            classname="p-1 fill-current stroke-current inline-block h-10 text-white"
          />
          <span
            class="sr-only"
          >
            Houzz
          </span>
        </a>
      </div>
    </div>
    <div
      class="bg-gray-darkest px-2 py-6 text-xs text-center text-gray-light"
    >
      © 2020 Bluebird Cabinetry & Design LLC. All Rights Reserved.
    </div>
  </div>
</DocumentFragment>
`;
